Plugin/Customer/Model/CustomerData.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Klaviyo\Reclaim\Plugin\Customer\Model;
+
+use Magento\Customer\CustomerData\Customer;
+use Magento\Customer\Helper\Session\CurrentCustomer;
+
+/**
+ * Class CustomerData
+ * @package Klaviyo\Reclaim\Plugin\Customer\Model
+ */
+class CustomerData
+{
+    /**
+     * @var CurrentCustomer
+     */
+    private $currentCustomer;
+
+    /**
+     * CustomerData constructor.
+     * @param CurrentCustomer $currentCustomer
+     */
+    public function __construct(
+        CurrentCustomer $currentCustomer
+    ) {
+        $this->currentCustomer = $currentCustomer;
+    }
+
+    /**
+     * @param Customer $subject
+     * @param $result
+     * @return mixed
+     */
+    public function afterGetSectionData(Customer $subject, $result)
+    {
+        if ($this->currentCustomer->getCustomerId()) {
+            $result['lastname'] = $this->currentCustomer->getLastname();
```

Comment:
can we add `firstname`?

Reply:
The first name is added by default magneto

see Magento\Customer\CustomerData\Customer

    public function getSectionData()
    {
        if (!$this->currentCustomer->getCustomerId()) {
            return [];
        }

        $customer = $this->currentCustomer->getCustomer();
        return [
            'fullname' => $this->customerViewHelper->getCustomerName($customer),
            'firstname' => $customer->getFirstname(),
            'websiteId' => $customer->getWebsiteId(),
        ];